### PR TITLE
SurvivalProbabilty Docs

### LINKED ISSUE #4247: Unclear documentation for SurvivalProbability function
## Expected behavior ##

The survival probability is the number of particles that have been continuously (or intermittently) in a group for a period of time $\tau$ divided by the number of particles in the group and the beginning of time window. This correlation is averaged over the number of time windows analyzed.

## Actual behavior ##

The existing documentation explains the SurvivalProbabilty function as being divided by the length of the simulation, implying units of inverse time.

## Code to reproduce the behavior ##

NA

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`): 2.6.0
- Which version of Python (`python -V`)? 3.9.16
- Which operating system? MacOS

### PR DESCRIPTION
**Pull request by jaclark5**
Fixes #4247

Changes made in this Pull Request:
 - Updated docs for SurvivalProbabilty function.

PR Checklist
------------
 - [NA] Tests?
 - [X] Docs?
 - [X] CHANGELOG updated?
 - [X] Issue raised/referenced?

## Developers certificate of origin
- [ ] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4248.org.readthedocs.build/en/4248/

### CODE REVIEW

**Review by hmacdope — APPROVED** [review]
Thanks for this @jaclark5, much better with the ensemble average marked clearly. :)

File: package/MDAnalysis/analysis/waterdynamics.py [hunk 1]
```diff
@@ -928,6 +931,8 @@ class SurvivalProbability(object):
        The `stop` keyword as passed to :meth:`SurvivalProbability.run` has now
        changed behaviour and will act in an `exclusive` manner (instead of it's
        previous `inclusive` behaviour),
+    .. versionchanged:: 2.7.0
+       Updated docs to align with discrete autocorrelation function.
```
Comment (RMeli) [inline package/MDAnalysis/analysis/waterdynamics.py:935]: @hmacdope Do we need to document docs changes/improvements as part of `versionchanged`?

Comment (hmacdope) [reply package/MDAnalysis/analysis/waterdynamics.py:935]: Normally I would say no, but in this case I would say makes sense as the previous expression wan't correct, this way if someone asks us why it was different previously we can point to this.

Comment (hmacdope) [reply package/MDAnalysis/analysis/waterdynamics.py:935]: @RMeli If you are happy you can approve and then I will merge.

Comment (hmacdope) [reply package/MDAnalysis/analysis/waterdynamics.py:935]: I am going to speak for @RMeli as he is busy atm.

**Pull request merged**